save-backend/src/main/kotlin/com/saveourtool/save/backend/service/TestSuitesSourceService.kt

```diff
@@ -156,8 +158,31 @@ class TestSuitesSourceService(
     fun fetch(
         testSuitesSource: TestSuitesSourceDto,
     ): Mono<EmptyResponse> = preprocessorWebClient.post()
-        .uri("/test-suites-sources/fetch")
-        .bodyValue(testSuitesSource)
+        .uri("/git/tag-list")
+        .bodyValue(testSuitesSource.gitDto)
         .retrieve()
-        .toBodilessEntity()
+        .bodyToMono<List<String>>()
+        .flatMapIterable { it }
+        .flatMap { tagName ->
+            preprocessorWebClient.post()
+                .uri("/test-suites-sources/fetch-from-tag?tagName={tagName}", tagName)
+                .bodyValue(testSuitesSource)
+                .retrieve()
+                .toBodilessEntity()
+        }
+        .collectList()
+        .flatMap {
+            preprocessorWebClient.post()
+                .uri("/git/default-branch-name")
+                .bodyValue(testSuitesSource.gitDto)
+                .retrieve()
+                .bodyToMono<String>()
+                .flatMap { branchName ->
```

Comment:
This `flatMap` can be moved out of the current lambda to avoid nesting

Reply:
thanks, moved